vendor/github.com/moby/buildkit/session/auth/authprovider/authprovider.go

```diff
@@ -28,17 +50,80 @@ type authProvider struct {
 	mu sync.Mutex
 }
 
+func (ap *authProvider) SetLogger(l progresswriter.Logger) {
```

Comment:
Need to connect this during build

Reply:
@tonistiigi  I had a look, but not sure how to handle this; the authProvider is added here; https://github.com/docker/cli/blob/88c6089300a82d3373892adf6845a4fed1a4ba8d/cli/command/image/build_buildkit.go#L179

To call `SetLogger`, looks like some ugly "interface" matching is needed, similar to https://github.com/moby/buildkit/blob/4177e7ca28c9918070a5ef30d4fa8a43d911e08d/cmd/buildctl/build.go#L269-L275

(curious; should `authProvider` be exported, or a `SetLogger()` interface be defined somewhere?)

```go
if s, ok := at.(interface {
	SetLogger(progresswriter.Logger)
}); ok {
	w := mw.WithPrefix("", false)
	s.SetLogger(func(s *client.SolveStatus) {
		w.Status() <- s
	})
```

However in the CLI, I don't think we have a `progresswriter.Logger` at this point in the code, as those are created in `doBuild` https://github.com/docker/cli/blob/88c6089300a82d3373892adf6845a4fed1a4ba8d/cli/command/image/build_buildkit.go#L276-L286


Perhaps I misunderstood how the logger should be set (can it be done at a later stage? if so, how do I get a reference to the `authProvider` later on?